src/dune_engine/diff_promotion.ml

```diff
@@ -179,3 +178,26 @@ let promote_files_registered_in_last_run files_to_promote =
   let db = load_db () in
   let db = do_promote db files_to_promote in
   dump_db db
+
+let diff_for_file (file : File.t) =
+  let msg = User_message.Annots.empty in
+  let a = Path.source file.dst in
+  let b = File.correction_file file in
+  Print_diff.get msg a b
+
+let filter_db files_to_promote db =
+  match files_to_promote with
+  | All -> db
+  | These (files, on_missing) ->
+    List.filter_map files ~f:(fun file ->
+        let r =
+          List.find db ~f:(fun (f : File.t) -> Path.Source.equal f.dst file)
+        in
+        if Option.is_none r then on_missing file;
+        r)
+
+let display files_to_promote =
+  let open Fiber.O in
+  let files = load_db () |> filter_db files_to_promote |> List.rev in
```

Comment:
Why do we need `List.rev`?

Reply:
That was to return in the right order. Instead I'm now mapping in parallel and reducing into a map.